flink-runtime/src/test/java/org/apache/flink/runtime/highavailability/FileSystemJobResultStoreFileOperationsTest.java

```diff
@@ -54,11 +58,19 @@ public class FileSystemJobResultStoreFileOperationsTest {
 
     private Path basePath;
 
+    private ExecutorService ioExecutor;
+
     @BeforeEach
     public void setupTest() throws IOException {
         basePath = new Path(temporaryFolder.toURI());
+        ioExecutor = Executors.newSingleThreadExecutor();
```

Comment:
I guess, for the test cases in this class, the `directExecutor` is good enough (rather than creating a new thread). You could also use `ManuallyTriggeredScheduledExecutor` if you want to verify delay in the execution.

Reply:
Fixed.